pylint/checkers/utils.py

```diff
@@ -1963,3 +1963,36 @@ def is_hashable(node: nodes.NodeNG) -> bool:
         return False
     except astroid.InferenceError:
         return True
+
+
+def is_augassign(node: nodes.NodeNG, parent: nodes.NodeNG) -> bool:
+    """
+    Determine if the node is being assigned to a modified version of
+    itself, aka an augmented assignment, but without a AugAssign being used.
+
+    For example,
+        obj.value = 1 + obj.value
+    is the same as
+        obj.value += 1
+    and this function is intended to detect the first case.
+    """
+    if not isinstance(node, nodes.AssignAttr):
+        return False
+
+    children = [x for x in parent.get_children()]
+    binops = [x for x in children if isinstance(x, nodes.BinOp)]
```

Comment:
As children is only used in the second list  comprehension, imbricating both list comprehension would be more efficient. 

I'm on mobile so I can't check get_children api but I wouldn't be surprised if you can get all the children of a certain type directly.

Reply:
I don't see a way to get children of a certain type https://pylint.pycqa.org/projects/astroid/en/latest/api/astroid.nodes.html#astroid.nodes.AugAssign.get_children
but will def combine the list comps

Reply:
Maybe node_of_classes but I'm not sure it's only the children that are returned. Sorry for suggesting things that I can't test myself right now, I'm on mobile.